chrome/content/zotero/xpcom/style.js

```diff
@@ -190,7 +190,26 @@ Zotero.Styles = new function() {
 		}
 		
 		return _styles[id] || false;
-	}
+	};
+	
+	this.getAsync = Zotero.Promise.coroutine(function* (id, skipMappings, download=true) {
+		yield this.init();
+		var style = this.get(id, skipMappings);
+		
+		if (!style && download) {
+			try {
+				// Don't wait too long
+				Zotero.debug(`Fetching required style ${id} asynchronously`);
+				let styleText = yield Zotero.File.getContentsFromURLAsync(id, {timeout: 5000});
+				yield Zotero.Styles.install(styleText, id, true);
+				style = this.get(id, skipMappings);
+			} catch (e) {
+				Zotero.debug(`Failed to retrieve style ${id}`);
+				Zotero.debug(e);
+			}	
+		}
+		return style;
+	});
```

Comment:
I don't think this should be part of a `getAsync()`. There are lots of things that are `*Async` in Zotero now, and they don't generally kick off HTTP requests. So I think we should make this more explicit, with a separate function (`fetch` or `download` or something) that explicitly makes a request if a `get()` fails. (I also don't think this is likely to be generally necessary functionality — it's pretty specific to the case of an embedded style in a stored document.)